src/controllers/controllermanager.cpp

```diff
@@ -131,7 +131,13 @@ ControllerManager::ControllerManager(UserSettingsPointer pConfig)
 
 ControllerManager::~ControllerManager() {
     emit requestShutdown();
-    m_pThread->wait();
+    // Some object (such as controller screen scheduler) runs in the GUI thread
+    // and will need to teardown before the Controller thread join the main
+    // thread. This is why we explicitly let the event loop run while we wait
+    // for the thread to join.
+    while (!m_pThread->wait(100)) {
+        QCoreApplication::processEvents(QEventLoop::AllEvents);
+    }
```

Comment:
Agreed, this is introduced with the latest single thread rendering (see my comment [here](https://github.com/mixxxdj/mixxx/pull/11407#issuecomment-2094501329)). AFAIK, they are no solution due to the chain `Core(main thread) -> Controller(Controller thread) -> Screen rendering (main thread)`, or perhaps you have an idea on how to deal with it?
Otherwise, Happy to revert that single threaded version which doesn't bring much value IMO.

Reply:
I don't quite understand. what chain are you referring too? order of destructors, thread-existing order? call-chain?

Reply:
but yeah. unless the single-threaded approach avoids some critical bugs, I don't see much value.

Reply:
Both order of destructors, and thread-existing order (in brackets).
Cool I will undo the change then. I guess we might have to revisit if when want to support more RHI in the future.

Reply:
Because QRhi doesn't support this kind of multi-threading?

Reply:
Yes, with Qt6, it appears that multi threading rendering support had a few regression. Qt even removed the multi threaded example they had for it.